src/Wrapper.js

```diff
@@ -158,10 +159,11 @@ class Wrapper extends React.PureComponent {
   }
 
   handleClosePreferences = () => {
-    this.setState({ preferencesOpened: false })
+    window.location.hash = getAppPath(this.props.locator)
   }
-  handleOpenPreferences = () => {
-    this.setState({ preferencesOpened: true })
+  handleOpenPreferences = path => {
+    const appPath = getAppPath(this.props.locator)
+    window.location.hash = `${appPath}${GLOBAL_PREFERENCES_QUERY_PARAM}${path}`
```

Comment:
I was thinking about doing something like this, where a more generic `getPath` would generate a path based on a locator + modifiers, and `parsePath()` would generate a path from a locator:

```
path => parsePath() => locator
locator => getPath() => path
```

It could look like this:

```js
// Updating the path
window.location.hash = getPath(
  // the current locator
  locator,
  {
    // modifiers
    preferences: 'home',
  }
)

// The other way wouldn’t change
const locator = parsePath(path)
```

That way the path would always be generated from the same function, that would take care of any case we add in the future.

Then, on top of this, maybe we could add a hook that would take care of storing the state of `locator` so we don’t need to pass it around, and updating `window.location.hash`?

/cc @sohkai @2color

Reply:
Can we take this out of scope for this PR? This ends up adding work and back and forth to what was a "well defined task"

Reply:
cc @luisivan 👆

Reply:
Of course. It’s more about discussing ideas in context than asking for these changes to be made in this PR specifically.